File: 2022/24/java/Main.java
import java.io.IOException;

public class Main {
	// --- Constants ---
	private static final String INPUT_FILE = "../input/input.txt";

	public static void main(String[] args) throws IOException {
		// --- Puzzle 1 ---
		Map map = new Map(INPUT_FILE);
		Point start = new Point(1, 0);
		Point end = new Point(map.getWidth() - 2, map.getHeight() - 1);
		int time = map.search(start, end, 0);
		System.out.format("1. Minutes to reach the goal: %,d%n", time);

		// --- Puzzle 2 ---
		time = map.search(end, start, time);
		time = map.search(start, end, time);
		System.out.format("2. Minutes to reach the goal, go back to the start, then reach the goal again: %,d%n", time);
	}
}

File: 2022/24/rust/src/main.rs
use rust::Map;

// --- Constants ---
const INPUT_FILE: &'static str = "../input/input.txt";

// --- Main function ---
fn main() -> Result<(), Box<dyn std::error::Error>> {
    // --- Read and parse the input file ---
    let mut map = Map::load(INPUT_FILE)?;
    let start = (1, 0);
    let end = (map.width - 2, map.height - 1);

    // --- Puzzle 1 ---
    let mut time = map.search(&start, &end, 0);
    println!("1. Minutes to reach the goal: {}", time);

    // --- Puzzle 2 ---
    time = map.search(&end, &start, time);
    time = map.search(&start, &end, time);
    println!("2. Minutes to reach the goal, go back to the start, then reach the goal again: {}", time);

    Ok(())
}
